kolibri/content/api.py

```diff
@@ -47,7 +48,20 @@ def cache_forever(some_func):
 
     def wrapper_func(*args, **kwargs):
         response = some_func(*args, **kwargs)
-        patch_response_headers(response, cache_timeout=cache_timeout)
+        # This caching has the unfortunate effect of also caching the dynamically
+        # generated filters for recommendation, this quick hack checks if
+        # the request is any of those filters, and then applies less long running
+        # caching on it.
+        timeout = cache_timeout
+        try:
+            request = args[0]
```

Comment:
if we get an `IndexError`, can we guarantee the request object does not exist in the kwargs?

Reply:
Hrm, true - probably not. I stopped here because in manual testing the args was sufficient. Will update.

Reply:
Is the idea that this won't work if somehow request is passed as a keyword?

```python
wrapper_func(request=request)
```

Is this even possible?